felix/netlinkshim/netlink.go

```diff
@@ -47,6 +49,156 @@ type Interface interface {
 	NeighDel(a *netlink.Neigh) error
 }
 
+type RealNetlink struct {
+	nlHandle *netlink.Handle
+}
+
 func NewRealNetlink() (Interface, error) {
-	return netlink.NewHandle(syscall.NETLINK_ROUTE)
+	nlHandle, err := netlink.NewHandle(syscall.NETLINK_ROUTE)
+	return &RealNetlink{
```

Comment:
Prefer to return `nil, err` if there's an error.  Makes incorrect use fail quicker.